src/views/PurchaseOrder.vue

```diff
@@ -133,11 +133,20 @@ export default defineComponent({
         return !this.fieldMappings[id] ? id : this.generateUniqueMappingPrefId();
       },
       saveMapping() {
-        if (this.mappingName) {
+        if (this.mappingName && this.file && this.areAllFieldsSelected()) {
+          showToast(translate("Mapping saved successfully"));
           const mappingPrefId = this.generateUniqueMappingPrefId();
           this.store.dispatch('user/updateFieldMappings', { mappingPrefId, mappingPrefName: this.mappingName, mappingPrefValue: JSON.parse(JSON.stringify(this.fieldMapping)) })
         } else {
-          showToast(translate("Enter mapping name"));
+          if(!this.mappingName) {
```

Comment:
We could add these conditions at the top and return control, if they are fulfilled. The code if then doesn't need any if else